boards/imix/src/test/ipv6_lowpan_test.rs

```diff
@@ -482,118 +481,140 @@ fn ipv6_check_receive_packet(
     unsafe {
```

Comment:
Is this `unsafe` still necessary now that the `ptr::read` have been removed?

I also want to make an observation that is not really related to this pull-request. In general I don't think having such a broad `unsafe` block is a good pattern:
- Either the `unsafe` part is located in some specific line of code (`ptr::read`), and should only cover this line, to avoid allowing unforeseen unsafe operations elsewhere in the block of code by mistake.
- Or the whole function is `unsafe`, pushing the callers to prove it's indeed safe to call it.
- In either case, there should be some comment or documentation explaining why this `unsafe` block exists.

But wrapping the whole body of a function in an `unsafe` block without marking the function as `unsafe` seems sneaky and IMO is the recipe for ending up with undefined behavior.

Reply:
The unsafe is still needed because the function relies on a bunch of accesses of global static mut variables (it could be scoped down a little bit now but not much). 

I completely agree with your comments regarding when it is appropriate to use unsafe, and agree that this test does not set a good example there. But I am a little reluctant to refactor this whole test when it is just a test for a single board that is not used elsewhere and will not be included in normal use of Tock. Either way such a refactoring is outside the scope of this PR.